File: code_samples/go/web_development_validation.go
package main

import (
    "fmt"
)

// Web Development: Validation
// AI/ML Training Sample

type Validation struct {
    Data string
}

func NewValidation() *Validation {
    return &Validation{
        Data: "",
    }
}

func (s *Validation) Process(input string) {
    s.Data = input
}

func (s *Validation) Validate() bool {
    return len(s.Data) > 0
}

func (s *Validation) GetData() string {
    return s.Data
}

func main() {
    instance := NewValidation()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}

File: code_samples/perl/web_development_validation.pl
#!/usr/bin/perl
# Web Development: Validation
# AI/ML Training Sample

package Validation;
use strict;
use warnings;

sub new {
    my $class = shift;
    my $self = {
        data => '',
    };
    bless $self, $class;
    return $self;
}

sub process {
    my ($self, $input) = @_;
    $self->{data} = $input;
}

sub get_data {
    my $self = shift;
    return $self->{data};
}

sub validate {
    my $self = shift;
    return length($self->{data}) > 0;
}

# Example usage
my $instance = Validation->new();
$instance->process("example");
print "Data: " . $instance->get_data() . "\n";
print "Valid: " . ($instance->validate() ? "true" : "false") . "\n";

1;
